src/test/scala/SchemaModuleExamples.scala

```diff
@@ -0,0 +1,158 @@
+package scalaz
+
+package schema
+
+import testz._
+import monocle._
+import monocle.macros._
+
+object SchemaModuleExamples {
+
+  val jsonModule = new SchemaModule {
+    type Prim[A]       = JsonSchema.Prim[A]
+    type ProductTermId = String
+    type SumTermId     = String
+  }
+
+  final case class Person(name: String, role: Option[Role])
+  sealed trait Role
+  final case class User(active: Boolean)       extends Role
+  final case class Admin(rights: List[String]) extends Role
+
+  object Person {
+    val name = Getter[Person, String](_.name)
+
+    def setRole(r1: Role): Person => Person = (p: Person) => p.copy(role = Some(r1))
+    val role                                = monocle.Optional[Person, Role](_.role)(setRole(_))
+
+    val user   = GenPrism[Role, User]
+    val admin  = GenPrism[Role, Admin]
+    val active = Getter[User, Boolean](_.active)
+    val rights = Getter[Admin, List[String]](_.rights)
+
+  }
+
+  object PersonSchema extends Schema[Person] {
+
+    def apply(module: SchemaModule): module.Schema[Person] = {
+      import module._
+      record[Person](
+        essentialField[Person, String](
+          "name",
```

Comment:
While is possible to access `SchemaModule` functions, there is no type parameter to peek into `ProducTermId` to know what's its type. Causing a type mismatch error.